### DISCUSSION TITLE
Is FastDDS v2.14.1 backwards compatible with FastDDS v2.1.4?

### DISCUSSION THREAD

**Discussion by gstangel** [eProsima/Fast-DDS#4929] [Q&A]
Hi there, this is a general question about backwards compatibility.  I am developing a distributed system based around ROS2 Foxy, which uses FastDDS v2.1.4 under the hood.  I would like to leverage some newer features in v2.14.1, like the ability to launch a discovery server from a XML file, so I built FastDDS from source.  I an unable to get the Discovery Server, running on v2.14.1, working with the ROS2 Foxy Nodes, running v2.1.4.  I was wondering if anyone would be able to point me in a direction that would define what versions of FastDDS are compatible with each other.  I am able to get the Discovery Server working when it is running at the same version as the ROS2 Foxy Nodes (v2.1.4).

Thanks!

**Comment by Mario-DL** [comment]
Hi @gstangel,
Thanks for using Fast DDS. In general terms we cannot guarantee that the behavior and ABI of `v2.14.1` is compatible with `v2.1.4` (that was deprecated a year ago) in all its features, what we can assure is that there is no API breaks within these two ([VERSIONING](https://github.com/eProsima/Fast-DDS/blob/master/VERSIONING.md)).

Anyways, please ensure that the client and server settings are correct (address, port and index). For further documentation you can refer to ([Discovery Server](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery_server.html))

**Comment by devrite** [comment]
I face a similar issue I use humble (2.6.8). If I build the DDSRouter WS it also builds it's own dds libs. If I launch the discovery server via ddsrouter or via `fastdds discoery` from the ws directly I can not get humble to connect to that router. If I use the humble bundled discovery server via `fastdds discoery` it works.

**Reply by devrite** [reply]
A workaround for my use case, which seems to work, is to start the dds discovery server externally bundled with ros and use a dds client in ddsrouter to connect to it. Using the current fast dds api would require some backports in rmw and some or backport rmw of rolling to humble.
